src/upload/mod.rs

```diff
@@ -263,6 +265,75 @@ pub async fn upload_package_to_anaconda(
     Ok(())
 }
 
+/// Uploads a package to a channel in an S3 bucket.
+pub async fn upload_package_to_s3(
+    storage: &AuthenticationStorage,
+    channel: Url,
+    endpoint_url: Option<Url>,
+    region: Option<String>,
+    force_path_style: Option<bool>,
+    package_files: &Vec<PathBuf>,
+) -> miette::Result<()> {
```

Comment:
we had some internal discussions whether we should reindex directly after uploading or leave that to `rattler-index s3`. for a proper package server i think having a cron job that runs `rattler-index s3` is still necessary to avoid potential race conditions when uploading two files at once.
i don't have any strong opinion on whether we should include it here.

Reply:
If we want to include it here, it's still beyond the scope of this PR as we don't have `rattler-index s3` yet. I'd go ahead and implement `rattler-index s3` next and then we can discuss if it makes sense to run it here or not.

Reply:
No I think these are two parallel discussions.
For me it would be fine to first merge this and then afterwards adjust the uploading here if we decide we want to reindex after uploading.